modules/apps/account/account-test/src/testIntegration/java/com/liferay/account/service/test/AccountRoleLocalServiceTest.java

```diff
@@ -306,6 +312,69 @@ public void testGetAccountRolesMultipleAccountEntries() throws Exception {
 		Assert.assertArrayEquals(expectedRoleIds, actualRoleIds);
 	}
 
+	@Test
+	public void testSearchAccountRoles() throws Exception {
+		String keywords = RandomTestUtil.randomString();
+
+		AccountRole accountRole = _addAccountRole(
+			_accountEntry1.getAccountEntryId(),
+			keywords + RandomTestUtil.randomString());
+
+		_addAccountRole(
+			_accountEntry1.getAccountEntryId(), RandomTestUtil.randomString());
+
+		_addAccountRole(
+			_accountEntry2.getAccountEntryId(),
+			keywords + RandomTestUtil.randomString());
+
+		BaseModelSearchResult<AccountRole> baseModelSearchResult =
+			AccountRoleLocalServiceUtil.searchAccountRoles(
+				_accountEntry1.getAccountEntryId(), StringPool.BLANK,
+				QueryUtil.ALL_POS, QueryUtil.ALL_POS, null);
+
+		Assert.assertEquals(2, baseModelSearchResult.getLength());
+
+		baseModelSearchResult = AccountRoleLocalServiceUtil.searchAccountRoles(
+			_accountEntry1.getAccountEntryId(), keywords, QueryUtil.ALL_POS,
+			QueryUtil.ALL_POS, null);
+
+		Assert.assertEquals(1, baseModelSearchResult.getLength());
+
+		List<AccountRole> accountRoles = baseModelSearchResult.getBaseModels();
+
+		Assert.assertEquals(accountRole, accountRoles.get(0));
+	}
+
+	@Test
+	public void testSearchAccountRolesWithPagination() throws Exception {
+		String keywords = RandomTestUtil.randomString();
+
+		for (int i = 0; i < 5; i++) {
+			_addAccountRole(_accountEntry1.getAccountEntryId(), keywords + i);
+		}
+
+		BaseModelSearchResult<AccountRole> baseModelSearchResult =
+			AccountRoleLocalServiceUtil.searchAccountRoles(
+				_accountEntry1.getAccountEntryId(), keywords, 0, 2, null);
+
+		List<AccountRole> accountRoles = baseModelSearchResult.getBaseModels();
+
+		Assert.assertEquals(accountRoles.toString(), 2, accountRoles.size());
+		Assert.assertEquals(_accountRoles.subList(0, 2), accountRoles);
```

Comment:
I think we should also add an assertion that the total hits size is larger than the paginated size. In this case, wouldn't `baseModelSearchResult` yield a count of 5?

Reply:
yes! I was going to add it and somehow forgot about it. Will update the test to include that. Thanks!